wire.md

```diff
@@ -0,0 +1,1008 @@
+
+`post_type` MUST be set to `1`.
+
+A client interpreting this post MUST only perform a local deletion of the
+referenced posts if the author (`post.public_key`) matches the author of the
+post to be deleted (i.e. only the user who authored a post may delete it).
+
+`num_deletions` MUST be set to 1 or greater.
+
+#### 6.2.4 `post/info`
+
+Set public information about one's self.
+
+field        | type               | desc
+-------------|--------------------|-------------------------
+`key1_len`   | `varint`           | length of the first key to set, in bytes
+`key1`       | `u8[key1_len]`     | name of the first key to set (UTF-8)
+`value1_len` | `varint`           | length of the first value to set, belonging to `key1`, in bytes
+`value1`     | `u8[value_len]`    | value of the first key:value pair
+...          |                    |
+`keyN_len`   | `varint`           | length of the Nth key to set, in bytes
+`keyN`       | `u8[keyN_len]`     | name of the Nth key to set (UTF-8)
+`valueN_len` | `varint`           | length of the Nth value to set, belonging to `keyN`, in bytes
+`valueN`     | `u8[value_len]`    | value of the Nth key:value pair
+
+`post_type` MUST be set to `2`.
+
+Several key/value pairs MAY be set at once. A post MUST indicate it is done
+specifying pairs by setting the final `keyN_len` to 0.
+
+A `post/info` post is a complete description of a user's self-published
+information. A `post/info` post fully replaces a previously known version, and
+are not additive. If a key is set in one `post/info` but not the subsequent
+one, that key MUST be treated as being set to its default. See the table of
+keys/values below for each key's default value.
+
+Keys MUST be UTF-8 strings, and MUST be between 1 and 128 codepoints in length.
+
+A value field MUST NOT exceed 4096 bytes (4 kibibytes) in length.
+
+The valid bytes for a value depends on the key. See the table below.
+
+The following keys SHOULD be supported:
+
+key       | value format | desc
+----------|--------------|---------------------------------------
+`name`    | UTF-8        | the name this user wishes to be known as. Default value is "" (empty string).
+
+To save space, a client MAY discard older versions of these messages for a
+particular user.
+
+#### 6.2.5 `post/topic`
+
+Set a topic for a channel.
+
+field          | type               | desc
+---------------|--------------------|-------------------------------------------------------
+`channel_len`  | `varint`           | length of the channel's name, in bytes
+`channel`      | `u8[channel_len] ` | channel name (UTF-8)
+`topic_len`    | `varint`           | length of the topic field, in bytes
+`topic`        | `u8[topic_len] `   | topic content (UTF-8)
+
+`post_type` MUST be set to `3`.
+
+A `topic` field MUST be a valid UTF-8 string, between 0 and 512 codepoints. A
+topic of length zero MUST be considered as the current topic being cleared to
+the empty string, "".
```

Comment:
I think this is me trying to communicate to implementors that they ought interpret the empty string as a cleared key, which means they could display something different for the topic (e.g. "no topic set") instead of just the empty string.